### DISCUSSION TITLE
How to understand the abbr words in OCC code base?

### DISCUSSION THREAD

**Discussion by lavender-lee** [Open-Cascade-SAS/OCCT#133] [General]
Recently, I have been reading the code associated with approximation.  However, I am confused with a lot of abbreviation words (maybe it is not English).

For instance,

- [myNivCont](https://github.com/Open-Cascade-SAS/OCCT/blob/158ed439c48389bbba0f4f85f10902d9cba96bca/src/AppDef/AppDef_Variational.cxx#L114)

- [mySomTab](https://github.com/Open-Cascade-SAS/OCCT/blob/158ed439c48389bbba0f4f85f10902d9cba96bca/src/AdvApp2Var/AdvApp2Var_Iso.cxx#L579)

- [myDifTab](https://github.com/Open-Cascade-SAS/OCCT/blob/158ed439c48389bbba0f4f85f10902d9cba96bca/src/AdvApp2Var/AdvApp2Var_Iso.cxx#L582)

- [mySSP](https://github.com/Open-Cascade-SAS/OCCT/blob/158ed439c48389bbba0f4f85f10902d9cba96bca/src/AppDef/AppDef_LinearCriteria.cxx#L52)

- [NUPIL](https://github.com/Open-Cascade-SAS/OCCT/blob/158ed439c48389bbba0f4f85f10902d9cba96bca/src/AdvApprox/AdvApprox_ApproxAFunction.cxx#L478)

Niv - ??
Cont - constraint
Som - ??
Tab - table
Dif - ??
NUPIL - N + UPIL ?

**Comment by dpasukhi** [comment]
The abbreviations in the code can indeed be confusing. 
OCCT was written originally by France team, there are a lot of france terms.

Niv: This is short for "Niveau," which is French for "level."
Cont: As you correctly identified, this stands for "constraint."
Som: This is likely short for "Somme," which is French for "sum."
Tab: As you correctly identified, this stands for "table" or "array."
Dif: This is short for "Difference."
NUPIL:  have no idea

**Reply by lavender-lee** [reply]
@dbarbier Thanks for your detailed explanation.
